src/status.js

```diff
@@ -0,0 +1,40 @@
+let loggedIn = false;
+let statusContentScript = Promise.resolve();
+
+const loggedInScript = `
+document.querySelector('html').classList.remove('logged-out');
+document.querySelector('html').classList.add('logged-in');
+`;
+const loggedOutScript = `
+document.querySelector('html').classList.remove('logged-in');
+document.querySelector('html').classList.add('logged-out');
+`;
+
+function injectLoggedInStatus(status) {
+  statusContentScript = statusContentScript.then(async () => {
+    if (status === loggedIn) {
+      return;
+    }
+    loggedIn = status;
+    // register a content script to inject status classes in the future
+    await browser.contentScripts.register({
```

Comment:
will this replace the previously registered content script?

Reply:
No, I forgot the deregistration part...